app/models/message.rb

```diff
@@ -62,7 +62,14 @@ def clone
     return self unless prev.present?
 
     attrs = prev.attributes.slice(*clone_attributes)
-    Message.create!(attrs.merge({ repeated: true }))
+    message = Message.create!(attrs.merge({ repeated: true }))
+    message.clone_relations
+  end
+
+  def clone_relations
+    step_values.clone_step :gender, gender
```

Comment:
@radinreth we have to check based on the variable which marked as gender then getting that variable name, some cases marked_as gender variable name can be not :gender same as the location